metrics/core/src/main/java/org/hyperledger/besu/metrics/prometheus/PrometheusGauge.java

```diff
@@ -47,17 +47,20 @@ public PrometheusGauge(
 
   @Override
   public synchronized void labels(final DoubleSupplier valueSupplier, final String... labelValues) {
-    if (labelValues.length != labelNames.size()) {
-      throw new IllegalArgumentException(
-          "Label values and label names must be the same cardinality");
-    }
+    validateLabelsCardinality(labelValues);
     if (observationsMap.putIfAbsent(List.of(labelValues), valueSupplier) != null) {
       final String labelValuesString = String.join(",", labelValues);
       throw new IllegalArgumentException(
           String.format("A gauge has already been created for label values %s", labelValuesString));
     }
   }
 
+  @Override
+  public boolean isLabelsObserved(final String... labelValues) {
```

Comment:
Couldn't see where this new method is used? Is it in the other PR?

Reply:
It should be used in PeerTaskExecutor.executeAgainstPeer to add labels for tasks we haven't seen before